test/Microsoft.Dnx.Tooling.FunctionalTests/DnuRestoreTests2.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using Microsoft.Dnx.Testing;
+using Xunit;
+
+namespace Microsoft.Dnx.Tooling.FunctionalTests
+{
+    public class DnuRestoreTests2 : DnxSdkFunctionalTestBase
+    {
+        [Theory]
+        [MemberData(nameof(ClrDnxSdks))]
+        public void DnuRestore_DoesNotAllowProjectToSatisfyPackageDependency(DnxSdk sdk)
+        {
+            var solution = TestUtils.GetSolution<DnuRestoreTests2>(sdk, "DependencyTargets");
+            var bOutputPath = Path.Combine(solution.ArtifactsPath, "B");
+
+            // Build a package for B
+            sdk.Dnu.Restore(solution.GetProject("B")).EnsureSuccess();
```

Comment:
This is very similar to what TestUtils.CreateLocalFeed is doing in the new infrastructure. See if you can use it. I believe the only missing capability is specifying a dnu pack configuration.